Scripts/Editor/Tests/Entities/Transform/TransformUtilTests.cs

```diff
@@ -0,0 +1,4391 @@
+                TransformUtil.ConvertWorldToLocalScale(worldToLocal_zero, new float3(2, -2, 0)),
+                Is.EqualTo(new float3(float.PositiveInfinity, float.NegativeInfinity, 0))
+            );
+            Logger.Debug("END: Expected error messages.");
+
+
+            //TODO: #116 - Transforms with non-uniform scale operations are not currently supported.
+            // Tests with assertions commented out below would replace these tests.
+            LocalToWorld localToWorld_nonUniform = new LocalToWorld()
+            {
+                Value = float4x4.TRS(point_zero, quaternion.identity, new float3(1.5f, 3, 1))
+            };
+            float4x4 worldToLocal_nonUniform = math.inverse(localToWorld_nonUniform.Value);
+
+#if DEBUG
+            Logger.Debug("BEGIN: Expected error messages.");
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(localToWorld_nonUniform, point_one);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(localToWorld_nonUniform, point_two);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(localToWorld_nonUniform, -point_one);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(localToWorld_nonUniform, -point_two);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(localToWorld_nonUniform, point_zero);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(localToWorld_nonUniform, new float3(2, -2, 0));
+
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(worldToLocal_nonUniform, point_one);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(worldToLocal_nonUniform, point_two);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(worldToLocal_nonUniform, -point_one);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(worldToLocal_nonUniform, -point_two);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(worldToLocal_nonUniform, point_zero);
+            LogAssert.Expect(LogType.Error, s_NonUniformScaleError);
+            TransformUtil.ConvertWorldToLocalScale(worldToLocal_nonUniform, new float3(2, -2, 0));
+            Logger.Debug("END: Expected error messages.");
+#endif
+
+            // float3x3 worldToLocal_nonUniformRotation = new float3x3(TransformUtil.ConvertWorldToLocalRotation(worldToLocal_nonUniform, quaternion.identity));
```

Comment:
There are a few of these commented out sections. Should we just delete all the commented out ones?

Reply:
They were all supposed to have this comment block above but I see I missed some.
The idea is that the day you go to implement #116 you have ready made tests to start working with.

Reply:
Ah gotcha